DOCS-1145: Authenticate with api key (#1916)
viamrobotics/docs: docs/program/apis/data-client.md

To use the Viam data client API, you first need to instantiate a [`ViamClient`](https://python.viam.dev/autoapi/viam/app/viam_client/index.html#viam.app.viam_client.ViamClient) and then instantiate an [`DataClient`](https://python.viam.dev/autoapi/viam/app/data_client/index.html#viam.app.data_client.DataClient).
See the following example for reference.

> To find the api key and secret, go to [Viam app](https://app.viam.com/), and navigate to the bottom of your location's page.

Author: same question as in cloud.md:
```suggestion
Use the Viam CLI [to generate an api key for your organization](https://docs.viam.com/manage/cli/#create-an-organization-api-key).
```

To use the Viam data client API, you first need to instantiate a [`ViamClient`](https://python.viam.dev/autoapi/viam/app/viam_client/index.html#viam.app.viam_client.ViamClient) and then instantiate an [`DataClient`](https://python.viam.dev/autoapi/viam/app/data_client/index.html#viam.app.data_client.DataClient).
See the following example for reference.

> To find the api key and secret, go to [Viam app](https://app.viam.com/), and navigate to the bottom of your location's page.

Reviewer: ```suggestion
To find the api key and secret, go to the [Viam app](https://app.viam.com/), and navigate to the bottom of your location's page.
```
But think this will likely need to be changed to `Security tab` if screenshot at top is accurate?

> To find the api key and secret, go to [Viam app](https://app.viam.com/), and navigate to the bottom of your location's page.

Author: ```suggestion
To find your api key and secret, go to [Viam app](https://app.viam.com/), and navigate to one of your robots' **Security** tab.
```

> For the URL, use the address of any of the robots in the location (found on the **Code sample** tab).

Author: ```suggestion
```

> For the URL, use the address of any of the robots in the location (found on the **Code sample** tab).

Author: ```suggestion
```
